File: src/main/java/com/qulix/shilomy/trainingtask/web/repository/EmployeeRepository.java
package com.qulix.shilomy.trainingtask.web.repository;

import com.qulix.shilomy.trainingtask.web.entity.EmployeeEntity;

import java.util.List;

public interface EmployeeRepository {
    void addEmployee(EmployeeEntity employee);
    void removeEmployee(EmployeeEntity employee);
    void updateEmployee(EmployeeEntity employee); // Think it as replace for set

    List<?> query(EmployeeSpecification specification);
}

File: src/main/kotlin/com/makshilo/trainingtask/repository/EmployeeRepository.kt
package com.makshilo.trainingtask.repository

import com.makshilo.trainingtask.model.Employee

interface EmployeeRepository {
  fun add(employee: Employee): Employee
  fun update(employee: Employee): Employee
  fun remove(id: Long)
  fun findById(id: Long): Employee?
  fun findAll(): List<Employee>
}
